// @aws-sdk/client-ivschat/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "ivschat",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class IvschatClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultIvschatHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let IvschatServiceException$1 = class IvschatServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, IvschatServiceException.prototype);
    }
};

let AccessDeniedException$1 = class AccessDeniedException extends IvschatServiceException$1 {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
};
let PendingVerification$1 = class PendingVerification extends IvschatServiceException$1 {
    name = "PendingVerification";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PendingVerification",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PendingVerification.prototype);
    }
};
let ResourceNotFoundException$1 = class ResourceNotFoundException extends IvschatServiceException$1 {
    name = "ResourceNotFoundException";
    $fault = "client";
    resourceId;
    resourceType;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
};
let ValidationException$1 = class ValidationException extends IvschatServiceException$1 {
    name = "ValidationException";
    $fault = "client";
    reason;
    fieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.reason = opts.reason;
        this.fieldList = opts.fieldList;
    }
};
let ConflictException$1 = class ConflictException extends IvschatServiceException$1 {
    name = "ConflictException";
    $fault = "client";
    resourceId;
    resourceType;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
};
let ServiceQuotaExceededException$1 = class ServiceQuotaExceededException extends IvschatServiceException$1 {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    resourceId;
    resourceType;
    limit;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
        this.limit = opts.limit;
    }
};
let ThrottlingException$1 = class ThrottlingException extends IvschatServiceException$1 {
    name = "ThrottlingException";
    $fault = "client";
    resourceId;
    resourceType;
    limit;
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
        this.limit = opts.limit;
    }
};
let InternalServerException$1 = class InternalServerException extends IvschatServiceException$1 {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
};

const _ADE = "AccessDeniedException";
const _CCT = "CreateChatToken";
const _CCTR = "CreateChatTokenRequest";
const _CCTRr = "CreateChatTokenResponse";
const _CE = "ConflictException";
const _CLC = "CreateLoggingConfiguration";
const _CLCR = "CreateLoggingConfigurationRequest";
const _CLCRr = "CreateLoggingConfigurationResponse";
const _CR = "CreateRoom";
const _CRR = "CreateRoomRequest";
const _CRRr = "CreateRoomResponse";
const _CT = "ChatToken";
const _CTA = "ChatTokenAttributes";
const _CWLDC = "CloudWatchLogsDestinationConfiguration";
const _DC = "DestinationConfiguration";
const _DLC = "DeleteLoggingConfiguration";
const _DLCR = "DeleteLoggingConfigurationRequest";
const _DM = "DeleteMessage";
const _DMR = "DeleteMessageRequest";
const _DMRe = "DeleteMessageResponse";
const _DR = "DeleteRoom";
const _DRR = "DeleteRoomRequest";
const _DU = "DisconnectUser";
const _DUR = "DisconnectUserRequest";
const _DURi = "DisconnectUserResponse";
const _FDC = "FirehoseDestinationConfiguration";
const _GLC = "GetLoggingConfiguration";
const _GLCR = "GetLoggingConfigurationRequest";
const _GLCRe = "GetLoggingConfigurationResponse";
const _GR = "GetRoom";
const _GRR = "GetRoomRequest";
const _GRRe = "GetRoomResponse";
const _ISE = "InternalServerException";
const _LCL = "LoggingConfigurationList";
const _LCS = "LoggingConfigurationSummary";
const _LLC = "ListLoggingConfigurations";
const _LLCR = "ListLoggingConfigurationsRequest";
const _LLCRi = "ListLoggingConfigurationsResponse";
const _LR = "ListRooms";
const _LRR = "ListRoomsRequest";
const _LRRi = "ListRoomsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _MRH = "MessageReviewHandler";
const _PV = "PendingVerification";
const _RL = "RoomList";
const _RNFE = "ResourceNotFoundException";
const _RS = "RoomSummary";
const _SDC = "S3DestinationConfiguration";
const _SE = "SendEvent";
const _SER = "SendEventRequest";
const _SERe = "SendEventResponse";
const _SQEE = "ServiceQuotaExceededException";
const _TE = "ThrottlingException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _UID = "UserID";
const _ULC = "UpdateLoggingConfiguration";
const _ULCR = "UpdateLoggingConfigurationRequest";
const _ULCRp = "UpdateLoggingConfigurationResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _URRp = "UpdateRoomRequest";
const _URRpd = "UpdateRoomResponse";
const _URp = "UpdateRoom";
const _VE = "ValidationException";
const _VEF = "ValidationExceptionField";
const _VEFL = "ValidationExceptionFieldList";
const _a = "attributes";
const _ar = "arn";
const _bN = "bucketName";
const _c = "client";
const _cT = "createTime";
const _cWL = "cloudWatchLogs";
const _ca = "capabilities";
const _dC = "destinationConfiguration";
const _dSN = "deliveryStreamName";
const _e = "error";
const _eN = "eventName";
const _f = "firehose";
const _fL = "fieldList";
const _fR = "fallbackResult";
const _h = "http";
const _hE = "httpError";
const _hQ = "httpQuery";
const _i = "id";
const _id = "identifier";
const _l = "limit";
const _lC = "loggingConfigurations";
const _lCI = "loggingConfigurationIdentifiers";
const _lCIo = "loggingConfigurationIdentifier";
const _lGN = "logGroupName";
const _m = "message";
const _mML = "maximumMessageLength";
const _mMRPS = "maximumMessageRatePerSecond";
const _mR = "maxResults";
const _mRH = "messageReviewHandler";
const _mRHU = "messageReviewHandlerUri";
const _n = "name";
const _nT = "nextToken";
const _r = "reason";
const _rA = "resourceArn";
const _rI = "resourceId";
const _rIo = "roomIdentifier";
const _rT = "resourceType";
const _ro = "rooms";
const _s = "state";
const _sDIM = "sessionDurationInMinutes";
const _sET = "sessionExpirationTime";
const _s_ = "s3";
const _se = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.ivschat";
const _t = "token";
const _tET = "tokenExpirationTime";
const _tK = "tagKeys";
const _ta = "tags";
const _u = "uri";
const _uI = "userId";
const _uT = "updateTime";
const n0 = "com.amazonaws.ivschat";
var ChatToken = [0, n0, _CT, 8, 0];
var UserID = [0, n0, _UID, 8, 0];
var AccessDeniedException = [
    -3,
    n0,
    _ADE,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException, AccessDeniedException$1);
var CloudWatchLogsDestinationConfiguration = [3, n0, _CWLDC, 0, [_lGN], [0]];
var ConflictException = [
    -3,
    n0,
    _CE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m, _rI, _rT],
    [0, 0, 0],
];
schema.TypeRegistry.for(n0).registerError(ConflictException, ConflictException$1);
var CreateChatTokenRequest = [
    3,
    n0,
    _CCTR,
    0,
    [_rIo, _uI, _ca, _sDIM, _a],
    [0, [() => UserID, 0], 64 | 0, 1, [() => ChatTokenAttributes, 0]],
];
var CreateChatTokenResponse = [
    3,
    n0,
    _CCTRr,
    0,
    [_t, _tET, _sET],
    [[() => ChatToken, 0], 5, 5],
];
var CreateLoggingConfigurationRequest = [
    3,
    n0,
    _CLCR,
    0,
    [_n, _dC, _ta],
    [0, () => DestinationConfiguration, 128 | 0],
];
var CreateLoggingConfigurationResponse = [
    3,
    n0,
    _CLCRr,
    0,
    [_ar, _i, _cT, _uT, _n, _dC, _s, _ta],
    [0, 0, 5, 5, 0, () => DestinationConfiguration, 0, 128 | 0],
];
var CreateRoomRequest = [
    3,
    n0,
    _CRR,
    0,
    [_n, _mMRPS, _mML, _mRH, _ta, _lCI],
    [0, 1, 1, () => MessageReviewHandler, 128 | 0, 64 | 0],
];
var CreateRoomResponse = [
    3,
    n0,
    _CRRr,
    0,
    [_ar, _i, _n, _cT, _uT, _mMRPS, _mML, _mRH, _ta, _lCI],
    [0, 0, 0, 5, 5, 1, 1, () => MessageReviewHandler, 128 | 0, 64 | 0],
];
var DeleteLoggingConfigurationRequest = [3, n0, _DLCR, 0, [_id], [0]];
var DeleteMessageRequest = [3, n0, _DMR, 0, [_rIo, _i, _r], [0, 0, 0]];
var DeleteMessageResponse = [3, n0, _DMRe, 0, [_i], [0]];
var DeleteRoomRequest = [3, n0, _DRR, 0, [_id], [0]];
var DisconnectUserRequest = [3, n0, _DUR, 0, [_rIo, _uI, _r], [0, [() => UserID, 0], 0]];
var DisconnectUserResponse = [3, n0, _DURi, 0, [], []];
var FirehoseDestinationConfiguration = [3, n0, _FDC, 0, [_dSN], [0]];
var GetLoggingConfigurationRequest = [3, n0, _GLCR, 0, [_id], [0]];
var GetLoggingConfigurationResponse = [
    3,
    n0,
    _GLCRe,
    0,
    [_ar, _i, _cT, _uT, _n, _dC, _s, _ta],
    [0, 0, 5, 5, 0, () => DestinationConfiguration, 0, 128 | 0],
];
var GetRoomRequest = [3, n0, _GRR, 0, [_id], [0]];
var GetRoomResponse = [
    3,
    n0,
    _GRRe,
    0,
    [_ar, _i, _n, _cT, _uT, _mMRPS, _mML, _mRH, _ta, _lCI],
    [0, 0, 0, 5, 5, 1, 1, () => MessageReviewHandler, 128 | 0, 64 | 0],
];
var InternalServerException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _se,
        [_hE]: 500,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InternalServerException, InternalServerException$1);
var ListLoggingConfigurationsRequest = [3, n0, _LLCR, 0, [_nT, _mR], [0, 1]];
var ListLoggingConfigurationsResponse = [
    3,
    n0,
    _LLCRi,
    0,
    [_lC, _nT],
    [() => LoggingConfigurationList, 0],
];
var ListRoomsRequest = [3, n0, _LRR, 0, [_n, _nT, _mR, _mRHU, _lCIo], [0, 0, 1, 0, 0]];
var ListRoomsResponse = [3, n0, _LRRi, 0, [_ro, _nT], [() => RoomList, 0]];
var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_rA], [[0, 1]]];
var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_ta], [128 | 0]];
var LoggingConfigurationSummary = [
    3,
    n0,
    _LCS,
    0,
    [_ar, _i, _cT, _uT, _n, _dC, _s, _ta],
    [0, 0, 5, 5, 0, () => DestinationConfiguration, 0, 128 | 0],
];
var MessageReviewHandler = [3, n0, _MRH, 0, [_u, _fR], [0, 0]];
var PendingVerification = [
    -3,
    n0,
    _PV,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(PendingVerification, PendingVerification$1);
var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m, _rI, _rT],
    [0, 0, 0],
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException, ResourceNotFoundException$1);
var RoomSummary = [
    3,
    n0,
    _RS,
    0,
    [_ar, _i, _n, _mRH, _cT, _uT, _ta, _lCI],
    [0, 0, 0, () => MessageReviewHandler, 5, 5, 128 | 0, 64 | 0],
];
var S3DestinationConfiguration = [3, n0, _SDC, 0, [_bN], [0]];
var SendEventRequest = [3, n0, _SER, 0, [_rIo, _eN, _a], [0, 0, 128 | 0]];
var SendEventResponse = [3, n0, _SERe, 0, [_i], [0]];
var ServiceQuotaExceededException = [
    -3,
    n0,
    _SQEE,
    {
        [_e]: _c,
        [_hE]: 402,
    },
    [_m, _rI, _rT, _l],
    [0, 0, 0, 1],
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaExceededException, ServiceQuotaExceededException$1);
var TagResourceRequest = [3, n0, _TRR, 0, [_rA, _ta], [[0, 1], 128 | 0]];
var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_e]: _c,
        [_hE]: 429,
    },
    [_m, _rI, _rT, _l],
    [0, 0, 0, 1],
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException, ThrottlingException$1);
var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_rA, _tK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
var UpdateLoggingConfigurationRequest = [
    3,
    n0,
    _ULCR,
    0,
    [_id, _n, _dC],
    [0, 0, () => DestinationConfiguration],
];
var UpdateLoggingConfigurationResponse = [
    3,
    n0,
    _ULCRp,
    0,
    [_ar, _i, _cT, _uT, _n, _dC, _s, _ta],
    [0, 0, 5, 5, 0, () => DestinationConfiguration, 0, 128 | 0],
];
var UpdateRoomRequest = [
    3,
    n0,
    _URRp,
    0,
    [_id, _n, _mMRPS, _mML, _mRH, _lCI],
    [0, 0, 1, 1, () => MessageReviewHandler, 64 | 0],
];
var UpdateRoomResponse = [
    3,
    n0,
    _URRpd,
    0,
    [_ar, _i, _n, _cT, _uT, _mMRPS, _mML, _mRH, _ta, _lCI],
    [0, 0, 0, 5, 5, 1, 1, () => MessageReviewHandler, 128 | 0, 64 | 0],
];
var ValidationException = [
    -3,
    n0,
    _VE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m, _r, _fL],
    [0, 0, () => ValidationExceptionFieldList],
];
schema.TypeRegistry.for(n0).registerError(ValidationException, ValidationException$1);
var ValidationExceptionField = [3, n0, _VEF, 0, [_n, _m], [0, 0]];
var __Unit = "unit";
var IvschatServiceException = [-3, _sm, "IvschatServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(IvschatServiceException, IvschatServiceException$1);
var LoggingConfigurationList = [1, n0, _LCL, 0, () => LoggingConfigurationSummary];
var RoomList = [1, n0, _RL, 0, () => RoomSummary];
var ValidationExceptionFieldList = [1, n0, _VEFL, 0, () => ValidationExceptionField];
var ChatTokenAttributes = [2, n0, _CTA, 8, 0, 0];
var DestinationConfiguration = [
    3,
    n0,
    _DC,
    0,
    [_s_, _cWL, _f],
    [
        () => S3DestinationConfiguration,
        () => CloudWatchLogsDestinationConfiguration,
        () => FirehoseDestinationConfiguration,
    ],
];
var CreateChatToken = [
    9,
    n0,
    _CCT,
    {
        [_h]: ["POST", "/CreateChatToken", 200],
    },
    () => CreateChatTokenRequest,
    () => CreateChatTokenResponse,
];
var CreateLoggingConfiguration = [
    9,
    n0,
    _CLC,
    {
        [_h]: ["POST", "/CreateLoggingConfiguration", 200],
    },
    () => CreateLoggingConfigurationRequest,
    () => CreateLoggingConfigurationResponse,
];
var CreateRoom = [
    9,
    n0,
    _CR,
    {
        [_h]: ["POST", "/CreateRoom", 200],
    },
    () => CreateRoomRequest,
    () => CreateRoomResponse,
];
var DeleteLoggingConfiguration = [
    9,
    n0,
    _DLC,
    {
        [_h]: ["POST", "/DeleteLoggingConfiguration", 204],
    },
    () => DeleteLoggingConfigurationRequest,
    () => __Unit,
];
var DeleteMessage = [
    9,
    n0,
    _DM,
    {
        [_h]: ["POST", "/DeleteMessage", 200],
    },
    () => DeleteMessageRequest,
    () => DeleteMessageResponse,
];
var DeleteRoom = [
    9,
    n0,
    _DR,
    {
        [_h]: ["POST", "/DeleteRoom", 204],
    },
    () => DeleteRoomRequest,
    () => __Unit,
];
var DisconnectUser = [
    9,
    n0,
    _DU,
    {
        [_h]: ["POST", "/DisconnectUser", 200],
    },
    () => DisconnectUserRequest,
    () => DisconnectUserResponse,
];
var GetLoggingConfiguration = [
    9,
    n0,
    _GLC,
    {
        [_h]: ["POST", "/GetLoggingConfiguration", 200],
    },
    () => GetLoggingConfigurationRequest,
    () => GetLoggingConfigurationResponse,
];
var GetRoom = [
    9,
    n0,
    _GR,
    {
        [_h]: ["POST", "/GetRoom", 200],
    },
    () => GetRoomRequest,
    () => GetRoomResponse,
];
var ListLoggingConfigurations = [
    9,
    n0,
    _LLC,
    {
        [_h]: ["POST", "/ListLoggingConfigurations", 200],
    },
    () => ListLoggingConfigurationsRequest,
    () => ListLoggingConfigurationsResponse,
];
var ListRooms = [
    9,
    n0,
    _LR,
    {
        [_h]: ["POST", "/ListRooms", 200],
    },
    () => ListRoomsRequest,
    () => ListRoomsResponse,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_h]: ["GET", "/tags/{resourceArn}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
var SendEvent = [
    9,
    n0,
    _SE,
    {
        [_h]: ["POST", "/SendEvent", 200],
    },
    () => SendEventRequest,
    () => SendEventResponse,
];
var TagResource = [
    9,
    n0,
    _TR,
    {
        [_h]: ["POST", "/tags/{resourceArn}", 200],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_h]: ["DELETE", "/tags/{resourceArn}", 200],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
var UpdateLoggingConfiguration = [
    9,
    n0,
    _ULC,
    {
        [_h]: ["POST", "/UpdateLoggingConfiguration", 200],
    },
    () => UpdateLoggingConfigurationRequest,
    () => UpdateLoggingConfigurationResponse,
];
var UpdateRoom = [
    9,
    n0,
    _URp,
    {
        [_h]: ["POST", "/UpdateRoom", 200],
    },
    () => UpdateRoomRequest,
    () => UpdateRoomResponse,
];

class CreateChatTokenCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "CreateChatToken", {})
    .n("IvschatClient", "CreateChatTokenCommand")
    .sc(CreateChatToken)
    .build() {
}

class CreateLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "CreateLoggingConfiguration", {})
    .n("IvschatClient", "CreateLoggingConfigurationCommand")
    .sc(CreateLoggingConfiguration)
    .build() {
}

class CreateRoomCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "CreateRoom", {})
    .n("IvschatClient", "CreateRoomCommand")
    .sc(CreateRoom)
    .build() {
}

class DeleteLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "DeleteLoggingConfiguration", {})
    .n("IvschatClient", "DeleteLoggingConfigurationCommand")
    .sc(DeleteLoggingConfiguration)
    .build() {
}

class DeleteMessageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "DeleteMessage", {})
    .n("IvschatClient", "DeleteMessageCommand")
    .sc(DeleteMessage)
    .build() {
}

class DeleteRoomCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "DeleteRoom", {})
    .n("IvschatClient", "DeleteRoomCommand")
    .sc(DeleteRoom)
    .build() {
}

class DisconnectUserCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "DisconnectUser", {})
    .n("IvschatClient", "DisconnectUserCommand")
    .sc(DisconnectUser)
    .build() {
}

class GetLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "GetLoggingConfiguration", {})
    .n("IvschatClient", "GetLoggingConfigurationCommand")
    .sc(GetLoggingConfiguration)
    .build() {
}

class GetRoomCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "GetRoom", {})
    .n("IvschatClient", "GetRoomCommand")
    .sc(GetRoom)
    .build() {
}

class ListLoggingConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "ListLoggingConfigurations", {})
    .n("IvschatClient", "ListLoggingConfigurationsCommand")
    .sc(ListLoggingConfigurations)
    .build() {
}

class ListRoomsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "ListRooms", {})
    .n("IvschatClient", "ListRoomsCommand")
    .sc(ListRooms)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "ListTagsForResource", {})
    .n("IvschatClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class SendEventCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "SendEvent", {})
    .n("IvschatClient", "SendEventCommand")
    .sc(SendEvent)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "TagResource", {})
    .n("IvschatClient", "TagResourceCommand")
    .sc(TagResource)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "UntagResource", {})
    .n("IvschatClient", "UntagResourceCommand")
    .sc(UntagResource)
    .build() {
}

class UpdateLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "UpdateLoggingConfiguration", {})
    .n("IvschatClient", "UpdateLoggingConfigurationCommand")
    .sc(UpdateLoggingConfiguration)
    .build() {
}

class UpdateRoomCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceChat", "UpdateRoom", {})
    .n("IvschatClient", "UpdateRoomCommand")
    .sc(UpdateRoom)
    .build() {
}

const commands = {
    CreateChatTokenCommand,
    CreateLoggingConfigurationCommand,
    CreateRoomCommand,
    DeleteLoggingConfigurationCommand,
    DeleteMessageCommand,
    DeleteRoomCommand,
    DisconnectUserCommand,
    GetLoggingConfigurationCommand,
    GetRoomCommand,
    ListLoggingConfigurationsCommand,
    ListRoomsCommand,
    ListTagsForResourceCommand,
    SendEventCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateLoggingConfigurationCommand,
    UpdateRoomCommand,
};
class Ivschat extends IvschatClient {
}
smithyClient.createAggregatedClient(commands, Ivschat);

const paginateListLoggingConfigurations = core.createPaginator(IvschatClient, ListLoggingConfigurationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListRooms = core.createPaginator(IvschatClient, ListRoomsCommand, "nextToken", "nextToken", "maxResults");

const ChatTokenCapability = {
    DELETE_MESSAGE: "DELETE_MESSAGE",
    DISCONNECT_USER: "DISCONNECT_USER",
    SEND_MESSAGE: "SEND_MESSAGE",
};
const ResourceType = {
    ROOM: "ROOM",
};
const ValidationExceptionReason = {
    FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
    OTHER: "OTHER",
    UNKNOWN_OPERATION: "UNKNOWN_OPERATION",
};
const CreateLoggingConfigurationState = {
    ACTIVE: "ACTIVE",
};
const FallbackResult = {
    ALLOW: "ALLOW",
    DENY: "DENY",
};
const LoggingConfigurationState = {
    ACTIVE: "ACTIVE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETE_FAILED: "DELETE_FAILED",
    DELETING: "DELETING",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
};
const UpdateLoggingConfigurationState = {
    ACTIVE: "ACTIVE",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AccessDeniedException = AccessDeniedException$1;
exports.ChatTokenCapability = ChatTokenCapability;
exports.ConflictException = ConflictException$1;
exports.CreateChatTokenCommand = CreateChatTokenCommand;
exports.CreateLoggingConfigurationCommand = CreateLoggingConfigurationCommand;
exports.CreateLoggingConfigurationState = CreateLoggingConfigurationState;
exports.CreateRoomCommand = CreateRoomCommand;
exports.DeleteLoggingConfigurationCommand = DeleteLoggingConfigurationCommand;
exports.DeleteMessageCommand = DeleteMessageCommand;
exports.DeleteRoomCommand = DeleteRoomCommand;
exports.DisconnectUserCommand = DisconnectUserCommand;
exports.FallbackResult = FallbackResult;
exports.GetLoggingConfigurationCommand = GetLoggingConfigurationCommand;
exports.GetRoomCommand = GetRoomCommand;
exports.InternalServerException = InternalServerException$1;
exports.Ivschat = Ivschat;
exports.IvschatClient = IvschatClient;
exports.IvschatServiceException = IvschatServiceException$1;
exports.ListLoggingConfigurationsCommand = ListLoggingConfigurationsCommand;
exports.ListRoomsCommand = ListRoomsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.LoggingConfigurationState = LoggingConfigurationState;
exports.PendingVerification = PendingVerification$1;
exports.ResourceNotFoundException = ResourceNotFoundException$1;
exports.ResourceType = ResourceType;
exports.SendEventCommand = SendEventCommand;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException$1;
exports.TagResourceCommand = TagResourceCommand;
exports.ThrottlingException = ThrottlingException$1;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateLoggingConfigurationCommand = UpdateLoggingConfigurationCommand;
exports.UpdateLoggingConfigurationState = UpdateLoggingConfigurationState;
exports.UpdateRoomCommand = UpdateRoomCommand;
exports.ValidationException = ValidationException$1;
exports.ValidationExceptionReason = ValidationExceptionReason;
exports.paginateListLoggingConfigurations = paginateListLoggingConfigurations;
exports.paginateListRooms = paginateListRooms;
